lnrpc/routerrpc/router_backend.go

```diff
@@ -768,6 +758,53 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 
 		payIntent.DestFeatures = features
 
+		// If this is an AMP payment, we must generate the initial
+		// randomness.
+		if rpcPayReq.Amp {
+			// First make sure the destination supports AMP, and
+			// that the payment address is set.
+			if features == nil || !features.HasFeature(lnwire.AMPOptional) {
+				return nil, fmt.Errorf("destination doesn't " +
+					"support AMP payments")
+			}
+
+			if len(rpcPayReq.PaymentAddr) == 0 {
```

Comment:
Yep, as before we add the invoice extensions, the payment addr can really be anything as there isn't any sort of expected association on the receiver's side.

Reply:
Ok, I thought this would be a "secret" that had to be communicated from the destination, but makes sense to create it automatically to support the keysend type use case 👍